public/app/features/alerting/unified/components/rule-editor/QueryRows.tsx

```diff
@@ -181,7 +181,7 @@ export class QueryRows extends PureComponent<Props> {
                       query={query}
                       onChangeQuery={this.onChangeQuery}
                       onRemoveQuery={this.onRemoveQuery}
-                      queries={queries}
```

Comment:
Was there any reason not to pass `expressions` here? Can we break something else by passing both `queries` and `expressions`?

Reply:
I _believe_ the logic for checking for duplicate refIDs is in a child component `QueryEditor`(?) but it doesn't really know how to render an expression. I'd still manually test this really well to understand the implications of passing the expression in there as well – I've done some manual testing and didn't find any bugs.

Reply:
I did a few tests too and I haven't found any issues